eth_utils/abi.py

```diff
@@ -42,27 +83,682 @@ def collapse_if_tuple(abi: Dict[str, Any]) -> str:
+
+        >>> from eth_utils import function_signature_to_4byte_selector
+        >>> function_signature_to_4byte_selector('myFunction()')
+        b'\xc3x\n:'
+    """
+    return keccak(text=function_signature.replace(" ", ""))[:4]
+
+
+def function_abi_to_4byte_selector(
+    function_abi: Union[ABIFunction, ABIError, ABIFallback, ABIReceive]
+) -> bytes:
+    r"""
+    Return the 4-byte function signature of the provided function ABI.
+
+    :param function_abi: Function ABI.
+    :type function_abi: `ABIFunction`
+    :return: 4-byte function signature.
+    :rtype: `bytes`
+
+    .. doctest::
+
+        >>> from eth_utils import function_abi_to_4byte_selector
+        >>> abi = {
+        ...   'type': 'function',
+        ...   'name': 'myFunction',
+        ...   'inputs': [],
+        ...   'outputs': []
+        ... }
+        >>> function_abi_to_4byte_selector(abi)
+        b'\xc3x\n:'
+    """
+    function_signature = abi_to_signature(function_abi)
     return function_signature_to_4byte_selector(function_signature)
 
 
 def event_signature_to_log_topic(event_signature: str) -> bytes:
+    r"""
+    Return the 32-byte keccak signature of the log topic for an event signature.
+
+    :param event_signature: String representation of the event name and arguments.
+    :type event_signature: `string`
+    :return: Log topic bytes.
+    :rtype: `bytes`
+
+    .. doctest::
+
+        >>> from eth_utils import event_signature_to_log_topic
+        >>> event_signature_to_log_topic('MyEvent()')
+        b'M\xbf\xb6\x8bC\xdd\xdf\xa1+Q\xeb\xe9\x9a\xb8\xfd\xedb\x0f\x9a\n\xc21B\x87\x9aO\x19*\x1byR\xd2'
+    """
     return keccak(text=event_signature.replace(" ", ""))
 
 
-def event_abi_to_log_topic(event_abi: Dict[str, Any]) -> bytes:
-    event_signature = _abi_to_signature(event_abi)
+def event_abi_to_log_topic(event_abi: ABIEvent) -> bytes:
+    r"""
+    Return the 32-byte keccak signature of the log topic from an event ABI.
+
+    :param event_abi: Event ABI.
+    :type event_abi: `ABIEvent`
+    :return: Log topic bytes.
+    :rtype: `bytes`
+
+    .. doctest::
+
+        >>> from eth_utils import event_abi_to_log_topic
+        >>> abi = {
+        ...   'type': 'event',
+        ...   'anonymous': False,
+        ...   'name': 'MyEvent',
+        ...   'inputs': []
+        ... }
+        >>> event_abi_to_log_topic(abi)
+        b'M\xbf\xb6\x8bC\xdd\xdf\xa1+Q\xeb\xe9\x9a\xb8\xfd\xedb\x0f\x9a\n\xc21B\x87\x9aO\x19*\x1byR\xd2'
+    """
+    event_signature = abi_to_signature(event_abi)
     return event_signature_to_log_topic(event_signature)
+
+
+def get_normalized_abi_arg_type(abi_element_param: Union[ABIComponent, str]) -> str:
+    """
+    Extract argument types from a function or event ABI parameter.
+
+    With tuple argument types, return a Tuple of each type.
+    Non-tuple types just return the type.
+
+    :param abi_element_param: ABI for the Function or Event parameter
+    :type abi_element_param: `ABIComponent`
```

Comment:
This can also take a `str`... is that true? If so, we should update this. If not, we should restrict the type in the function signature to just `ABIComponent`.

Reply:
I think the `param`, `arg`, `component` naming is inconsisent too. Thinking it could be named `get_normalized_abi_component_type` like:

```python
def get_normalized_abi_component_type(abi_component: Union[ABIComponent, str]) -> str:
```

The `str` argument is allowed so maybe the argument name should be something other than `abi_component`, what do you think?